src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -905,6 +905,17 @@ void QgsWmsCapabilities::parseLayer( QDomElement const & e, QgsWmsLayerProperty&
 
         parseStyle( e1, styleProperty );
 
+        for ( int i = 0; i < layerProperty.style.size(); ++i )
+        {
+          if ( layerProperty.style[i].name == styleProperty.name )
```

Comment:
The non-const version does a detach. And since while reviewing it's often not straightforward to see if something is used const or not I tend to prefer `.at()` which does not detach.